### ISSUE TITLE
DatePicker date formatting

### ISSUE THREAD

**Issue by resistorsoftware** [Shopify/polaris#346]
Shopify endpoints are looking for the formats:

created_at_min | Show orders created at or after date (format: **2014-04-25T16:15:47-04:00**).
created_at_max | Show orders created at or before date (format: **2014-04-25T16:15:47-04:00**).

The Polaris component returns something close.. but... not the same... 

**2018-02-28T05:00:00.000Z**

So if we pump Polaris data into API calls, is this expected to work? Why would the Polaris component pump formatting not suggested by the API endpoints?

The funny thing is, I cannot even use Shopify API Rest calls with dates and get filtered results that respect the dates... so I am kinda wondering aloud whether this Polaris Issue is an Issue amongst many, or just can be ignored and one date date filtering will work regardless. 

Any insight on this most appreciated.

**Comment by ry5n** [comment]
That’s a really good question. I’ve opened an internal issue with the relevant people. Stay tuned!

**Comment by resistorsoftware** [comment]
Here is another oddity... Suppose I want to pump your Polaris Datepicker into the API to get data, like orders. Well, your created_at_min and created_at_max fields are useful here, and they work as per they should (we hope)... 

So I present a Datepicker and the merchant chooses June 7, 2018, as both a start and end date for orders. I placed 500 orders with the Developer Tools today, and they are all June 7, 2018. Thing is, since Datepicker formats the time as per the time the merchant clicks the date... the time eliminates all orders from today, since the Datepicker tool sends time. So pumping that straight into an API call is useless. 

That makes this Datepicker somewhat useless without more concentrated formatting. I can expect to have to strip off time, and replace it with something like start 00:00:00 and end 23:59:59 which means I am now inserting some ad hoc code every single time I choose Datepicker. 

I have no sense of how you'd fix this other than to say, allow either Datepicker to come with some kind of formatting template, or loosen the rules on API filtering to allow just date portion through... 

Basically, if I want all orders from today alone, with Datepicker, I cannot just pump Datepicker dates to API... does not compute.

**Comment by tmlayton** [comment]
@resistorsoftware both date formats are valid [ISO 8601 time zone designator formats](https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators) and should work with the Shopify API. I ran a couple tests to confirm both formats worked.

    GET /admin/orders/count.json?updated_at_min=2018-02-28T05:00:00.000Z
    GET /admin/orders/count.json?updated_at_min=2014-04-25T16:15:47-04:00

As far as the results not respecting the dates, that would be an issue with the Shopify API itself and I would suggest posting in the [API forums](https://ecommerce.shopify.com/c/shopify-apis-and-technology).

And regarding the time issue and selecting a single day – the date picker is a naive component and uses the native `Date` object under the hood so this is a default behavior of the JavaScript `Date` and `Date#toISOString`. For example:

```js
const date = new Date(2018, 07, 06);
date.toISOString(); // -> "2018-08-06T07:00:00.000Z"
```

For now, the expectation is this is handled by the implementer in the callback.
